packages/perseus/src/widgets/radio/widget.jsx

```diff
@@ -443,28 +476,35 @@ class Radio extends React.Component<Props> {
             // Support legacy choiceState implementation
         }
         /* istanbul ignore if - props.values is deprecated */
-        // $FlowFixMe[prop-missing]
         if (props.values) {
             let noneOfTheAboveIndex = null;
             let noneOfTheAboveSelected = false;
 
-            const values = props.values.slice();
+            const choicesSelected = props.values.map(Boolean);
             const countChoices = props.countChoices;
             const numCorrect = props.numCorrect;
+            const valuesLength = props.values?.length ?? 0;
 
-            for (let i = 0; i < props.values.length; i++) {
+            for (let i = 0; i < valuesLength; i++) {
                 const index = props.choices[i].originalIndex;
-                values[index] = props.values[i];
+                if (
+                    index &&
+                    index < choicesSelected.length &&
+                    props.values &&
+                    i < props.values.length
+                ) {
```

Comment:
**suggestion:** I believe this can be simplified to
```
if (index < choicesSelected.length) {
```
`originalIndex` is a required property thanks to the new `RadioChoiceWithMetadata` and we know that `i < props.values.length` is true because of the condition in the `for` statement.

Reply:
We can probably drop the `if` completely b/c `choicesSelected` should have the same length as `props.values`.

Reply:
Dropped the if! XD